mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      var maxIdx = 0
+      var maxValue = if(indices(0) != 0) 0.0 else values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue < 0){
+        maxIdx = calcInactiveIdx(indices(0))
+        maxValue = 0
+      }
+      maxIdx
+    }
+  }
+
+  /**
+   * Calculates the first instance of an inactive node in a sparse vector and returns the Idx
+   * of the element.
+   * @param idx starting index of computation
+   * @return index of first inactive node or -1 if it cannot find one
+   */
+  private[SparseVector] def calcInactiveIdx(idx: Int): Int ={
```

Comment:
Yeah I was going back and forth on if I wanted to pass in a idx param or not. It would be nice in case we want to say find an inactive value after a given index but thats probably coding for the future which tends to be messy. I'll remove it for now and if anyone else has any other opinions we can go from there. 

I dunno I think I am just partial to recursive functions but I can give yours a try still. Really up for whatever best fits the spark code style etc.